src/main/java/org/opensearch/security/configuration/SecurityIndexSearcherWrapper.java

```diff
@@ -152,6 +152,11 @@ protected final boolean isBlockedProtectedIndexRequest() {
     }
 
     protected final boolean isBlockedSystemIndexRequest() {
+        boolean isSystemIndex = systemIndexMatcher.test(index.getName());
+        if (!isSystemIndex) {
+            return false;
+        }
```

Comment:
While this fixes the issue, the problem is also because [isPermittedOnSystemIndex](https://github.com/opensearch-project/security/blob/6b9762b17ac109c268b7cfc58ad42e39c4349615/src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java#L568-L583) doesn't check if the index being evaluated is a system index or not. Thoughts on fixing this in isPermittedOnSystemIndex?

Reply:
@shikharj05 
Great.
I will modify the `isPermittedOnSystemIndex` function to return true for indexes that are not system indexes.

Reply:
@shikharj05 
To check in Security Roles, it is necessary to pass the system index information. 
Although it is possible to have system index information within `SecurityRoles`, 
this information needs to be duplicated each time a `Filter` or `SecurityRoles` object is re-created.

For now, I have implemented it as shown here: https://github.com/opensearch-project/security/pull/4430/commits/cca772e2c3128353eca9713c3ac60756350e13f3. 
What do you think?

Reply:
okay, looking at changes - it's better to keep it here.